### PR TITLE
Unify naming of feature spec implementations

### PR DESCRIPTION
**Pull request by thecoop**
Use the same naming convention for feature spec implementations - `<area>Features`

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/features/FeaturesFeatures.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
  * Nodes that do not support features won't have this feature in its feature set,
  * so this can be checked without needing to look at the node version.
  */
-public class FeaturesSupportedSpecification implements FeatureSpecification {
+public class FeaturesFeatures implements FeatureSpecification {
```
Comment (ldematte) [inline server/src/main/java/org/elasticsearch/features/FeaturesFeatures.java:18]: This looks funny but I don't think there is a better alternative and it does stick to the schema after all. Hope no one will rename it thinking is a typo :)

**Review by ldematte — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/features/FeaturesFeatures.java (hunk 1 above)
Comment (thecoop) [reply server/src/main/java/org/elasticsearch/features/FeaturesFeatures.java:20]: I've added a javadoc highlighting the recursiveness of the definition

Comment (rjernst) [reply server/src/main/java/org/elasticsearch/features/FeaturesFeatures.java:20]: Even with the javadoc, this name is incredibly confusing. How about FeatureDetection or FeatureInfrastructure?

Comment (thecoop) [reply server/src/main/java/org/elasticsearch/features/FeaturesFeatures.java]: I've changed to `FeatureInfrastructureFeatures`

**Pull request merged**